sdk/python/feast/repo_operations.py

```diff
@@ -145,134 +141,102 @@ def apply_total(repo_config: RepoConfig, repo_path: Path, skip_source_validation
     from colorama import Fore, Style
 
     os.chdir(repo_path)
-    registry_config = repo_config.get_registry_config()
-    project = repo_config.project
+    store = FeatureStore(repo_path=str(repo_path))
+    project = store.project
     if not is_valid_name(project):
         print(
             f"{project} is not valid. Project name should only have "
             f"alphanumerical values and underscores but not start with an underscore."
         )
         sys.exit(1)
-    registry = Registry(
-        registry_path=registry_config.path,
-        repo_path=repo_path,
-        cache_ttl=timedelta(seconds=registry_config.cache_ttl_seconds),
-    )
+    registry = store.registry
     registry._initialize_registry()
     sys.dont_write_bytecode = True
     repo = parse_repo(repo_path)
     _validate_feature_views(repo.feature_views)
-    data_sources = [t.batch_source for t in repo.feature_views]
 
     if not skip_source_validation:
+        data_sources = [t.input for t in repo.feature_views]
         # Make sure the data source used by this feature view is supported by Feast
         for data_source in data_sources:
-            data_source.validate(repo_config)
+            data_source.validate(store.config)
 
-    # Make inferences
-    update_entities_with_inferred_types_from_feature_views(
-        repo.entities, repo.feature_views, repo_config
+    entities_to_keep, entities_to_delete = _tag_registry_entities_for_keep_delete(
+        project, registry, repo
+    )
+    views_to_keep, views_to_delete = _tag_registry_views_for_keep_delete(
+        project, registry, repo
     )
-    update_data_sources_with_inferred_event_timestamp_col(data_sources, repo_config)
-    for view in repo.feature_views:
-        view.infer_features_from_batch_source(repo_config)
```

Comment:
Is the type inference happening somewhere else?

Reply:
yes. it happens as part of the FeatureStore#apply